### DISCUSSION TITLE
new and need help! does it work with multiple needles?

### DISCUSSION THREAD

**Discussion by Joshua-Bettles** [inkstitch/inkstitch#1079] [General]
My company is looking at investing in a 15 needle machine and I've been tasked to find some good software to digitise the designs. How well does inkstitch deal with multiple needles? is it just a parameter that is enter or does it only work with one needle or is all the multiple needles managed and controlled by colour at the machine end

thanks

**Comment by lexelby** [comment]
Yup, it works great with multineedle machines.  I have a 9-needle.  Ink/Stitch sews objects in the order they're stacked in the SVG, with lower-stacked objects coming first.  If the color changes from one object to the next, Ink/Stitch automatically inserts a color change instruction.

On multineedle machines, when you load a design, it looks for the color change instructions and uses them to split the design into color blocks.  You assign each color block to a needle before you hit "go".

**Comment by wwderw** [comment]
It works just fine with multiple needs.  I run a Barudan C01 with a sequin attachment (can do that as well with Ink/Stitch, but that is more of a power user situation). And a 10 and 6 needle Brother.

I have found it best to deal with color blocks in terms of layers. I will have the source file as the top layer with opacity set down so I can see my digitizing.

Then I work from the bottom layer to the top.

I'll give my layers the title of something like "Grass - [Color that I am using]", if I do any detail, Ill have in a layer above that labeled "Grass Detail - [Some darker color to be used]".

Programs like Wilcom do this automatically as you use use different colors (or swap between already used colors) they automatically create color blocks and the user just has to reorganize the layers (if needed).  I find that using layers makes it easier to do edits and control the design better and also lockdown/hide areas that I may not want cluttering up my screen real estate.

I know beyond the scope of the question that you asked, I just thought it was worth mentioning as it does help with workflow, regardless if single or multi-needle that is a difference between a lot of the other programs out there.
